eclipse-theia/theia
# Notebook: Split cell command implementation

#### What it does


Implements a "Split cell" command. The split cell command splits the cell contents at the cursor position. After execution, the contents of the source cell contain the text before the cursor position and the new cell contain the text after the cursor position.


<img width="377" alt="Bildschirmfoto 2024-09-23 um 13 53 11" src="https://github.com/user-attachments/assets/da367992-012a-4045-894e-c05b51a1d6e3">

#### How to test


Create a notebook cell with some code inside.

<img width="438" alt="Bildschirmfoto 2024-09-23 um 13 56 46" src="https://github.com/user-attachments/assets/e7fc9529-9098-4d6c-a72e-e93848fbea8b">

Set the cursor e.g. on the second line and execute the "Split cell" toolbar action, or press `Ctrl+Shift+-`.

After execution:
- The source cell content should contain the text before the cursor position. 
- A new cell should be crated containing the text after the cursor  position.

<img width="436" alt="Bildschirmfoto 2024-09-23 um 13 57 03" src="https://github.com/user-attachments/assets/1a492f83-3845-4a87-8949-d468e25c47b1">

Note: I'm not sure if `ctrlShift-` will work on Windows, please check if possible.

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
